### PR TITLE
Add pointing interpolated values

### PR DESCRIPTION
**Pull request by morcuended**
Here is the modification to add non-empty gps_time and interpolate the pointing values event-wise from the values contained in the Drive log files.

Something weird happened: I thought I had made a fresh installation of the lstchain repo yesterday, following the instructions written en the README. However, apparently the ctapipe-io-lst package was not pointing to the last version v0.2 as it is specified in environment.yml. So some names within the LSTDriveContainer were not properly set. I had to reinstall it manually to make everything work.

Disclaimer: When adding the pointing interpolated values, the code becomes substantially slower. Maybe this could be optimized. Perhaps it is due to the large file size of the Drive log.

Please have a look and let me know if you have any comment/suggestion

### CODE REVIEW

File: lstchain/reco/dl0_to_dl1.py [hunk 1]
```diff
@@ -295,7 +294,39 @@ def r0_to_dl1(input_filename=get_dataset_path('gamma_test_large.simtel.gz'),
...
+                            utc_time = Time(datetime.utcfromtimestamp(ucts_timestamp * ns))
+                        else:
+                            start_ntp = event.lst.tel[telescope_id].svc.date
+                            ntp_time = start_ntp + event.r0.tel[telescope_id].trigger_time
+                            utc_time = Time(datetime.utcfromtimestamp(ntp_time))
+
+                        gps_time = utc_time.gps
+                        dl1_container.gps_time = gps_time
+
+                    if pointing_file_path:
```
Comment (morcuended) [inline lstchain/reco/dl0_to_dl1.py]: I just caught one error, this should be indented inside the `if not is_simu:` condition. I will fix it later on

**Comment by morcuended** [comment]
This is an example of how to run the script adding pointing information:

```
python lstchain_data_r0_to_dl1.py -f ~/LST-1.1.Run01388.0000.fits.fz \
    -conf config.json \
    -pedestal ~/pedestal_run1575_0000.fits \
    -calib ~/calibration_run1572.hdf5 \
    -pointing ~/drive_log/drive_log_19_09_19.txt
```
simply parsing the path to the drive log file. The problem at this moment is that one has to check first which drive report file should use because they usually mix several days.

Output:
Screenshot_2019-11-21_14-10-34

File: lstchain/reco/dl0_to_dl1.py (hunk 1 above)
Comment (morcuended) [reply lstchain/reco/dl0_to_dl1.py]: Fixed below

File: lstchain/reco/dl0_to_dl1.py [hunk 2]
```diff
@@ -295,7 +294,39 @@ def r0_to_dl1(input_filename=get_dataset_path('gamma_test_large.simtel.gz'),
...
+                    # one started to work around mid Nov 2019, so previous runs take
+                    # timing information from the TIB NTP counters.
+                    # This will be deprecated and modified back to the commented line
+                    # whenever GPS time is dumped to gps_time field.
+
+                    if not is_simu:
+                        ucts_timestamp = event.lst.tel[telescope_id].evt.ucts_timestamp
+ 
+                        if ucts_timestamp != 0:
+                            ns = 1e-9  # Convert nanosecs to secs
```
Comment (vuillaut) [inline lstchain/reco/dl0_to_dl1.py]: Using astropy units has been on the todo list for a long time.
Starting now would be great :)

File: lstchain/reco/dl0_to_dl1.py [hunk 3]
```diff
@@ -295,7 +294,39 @@ def r0_to_dl1(input_filename=get_dataset_path('gamma_test_large.simtel.gz'),
...
+                        else:
+                            start_ntp = event.lst.tel[telescope_id].svc.date
+                            ntp_time = start_ntp + event.r0.tel[telescope_id].trigger_time
+                            utc_time = Time(datetime.utcfromtimestamp(ntp_time))
+
+                        gps_time = utc_time.gps
+                        dl1_container.gps_time = gps_time
+
+                        if pointing_file_path:
+                            pointing_file = pointing_file_path
```
Comment (vuillaut) [inline lstchain/reco/dl0_to_dl1.py]: redundant

**Comment by morcuended** [comment]
> Looks overall good.
> I haven't tested with real data.
> I am working on getting a real data sample for unit tests, in the meantime, did you run it on real data and get a proper DL1 output ?

Yes, I tested it with one subrun of real data on my laptop I think getting proper output values.

**Comment by vuillaut** [comment]
@morcuended dl1ab test is failing.

**Comment by morcuended** [comment]
> @morcuended dl1ab test is failing.
> can you see on your laptop why?

I think it is because that branch was merged in the meanwhile and I did not pull it to my local repo. Will fix it and push again

**Comment by morcuended** [comment]
I do not understand why is failing though. The scripts I modified is not dependant on lstchain_mc_dl1ab.py. I pulled the changes on my local repo and ran pytest. It's failing too.

What should I do?

File: lstchain/reco/dl0_to_dl1.py (hunk 3 above)
Comment (morcuended) [reply lstchain/reco/dl0_to_dl1.py]: fixed

File: lstchain/reco/dl0_to_dl1.py (hunk 2 above)
Comment (morcuended) [reply lstchain/reco/dl0_to_dl1.py]: done

**Comment by rlopezcoto** [comment]
After pulling the master, did you merge it?
There should be some message of "merged master branch" after you push it and I do not see it.

**Comment by morcuended** [comment]
> After pulling the master, did you merge it?
> There should be some message of "merged master branch" after you push it and I do not see it.

Oh, I forgot that last step. I just pushed my branch again.

**Comment by morcuended** [comment]
Still failing

**Comment by rlopezcoto** [comment]
Last changes are not merged yet. Can you do:
 `git fetch upstream`
 `git merge upstream/master`
 `git push`

**Comment by vuillaut** [comment]
@morcuended weird it's not failing on my system, I'm investigating.

**Comment by morcuended** [comment]
```
(dev-lstchain) [dmorcuende@Toshiba cta-lstchain]$ git fetch upstream 
remote: Enumerating objects: 111, done.
remote: Counting objects: 100% (111/111), done.
remote: Compressing objects: 100% (45/45), done.
remote: Total 125 (delta 75), reused 96 (delta 65), pack-reused 14
Receiving objects: 100% (125/125), 20.56 KiB | 0 bytes/s, done.
Resolving deltas: 100% (77/77), completed with 12 local objects.
From https://github.com/cta-observatory/cta-lstchain
 * [new branch]      ctapipe_master -> upstream/ctapipe_master
 * [new branch]      master     -> upstream/master
 * [new branch]      production_scripts -> upstream/production_scripts
 * [new tag]         v0.1.0     -> v0.1.0

(dev-lstchain) [dmorcuende@Toshiba cta-lstchain]$ git merge upstream/master
Already up-to-date.

(dev-lstchain) [dmorcuende@Toshiba cta-lstchain]$ git push
Everything up-to-date

```

**Comment by vuillaut** [comment]
@morcuended actually, master is failing....

**Comment by morcuended** [comment]
> @morcuended actually, master is failing....

I see. However, when PR #222 was merged, all tests passed, right?

**Comment by vuillaut** [comment]
> > @morcuended actually, master is failing....
> 
> I see. However, when PR #222 was merged, all tests passed, right?

That's why I am a bit confused right now. Working on patch...

**Comment by vuillaut** [comment]
#224 should fix the issue.
You'll need to pull master again afterwards.

**Comment by rlopezcoto** [comment]
@morcuended please merge the new master, after #224 fix tests should pass.

**Comment by morcuended** [comment]
> @morcuended please merge the new master, after #224 fix tests should pass.

Just did it. I think tests are OK now. Thanks a lot, @vuillaut and @rlopezcoto

**Pull request merged**